python/ray/tests/test_client_terminate.py

```diff
@@ -105,21 +109,26 @@ class A:
         def f():
             time.sleep(1000)
 
-        actor = A.remote()
-        task_ref = f.remote()
+        class MetadataIsCorrectlyPassedException(Exception):
+            pass
 
         def mock_terminate(term, metadata):
-            raise Exception(metadata[0][0])
+            raise MetadataIsCorrectlyPassedException(metadata[0][0])
 
+        # Mock stub's Terminate method to raise an exception.
         ray.api.worker.server.Terminate = mock_terminate
 
-        with pytest.raises(Exception) as e:
+        # Verify the expected exception is raised with ray.kill.
+        with pytest.raises(MetadataIsCorrectlyPassedException) as e:
+            actor = A.remote()
             ray.kill(actor)
-        assert str(e.value) == "foo"
+        assert str(e.value) == "key"
 
-        with pytest.raises(Exception) as e:
+        # Verify the expected exception is raised with ray.cancel.
+        with pytest.raises(MetadataIsCorrectlyPassedException) as e:
+            task_ref = f.remote()
             ray.cancel(task_ref)
-        assert str(e.value) == "foo"
+        assert str(e.value) == "key"
```

Comment:
can you please add a tab before this to be below the with statement?

Reply:
nack
If I add a tab, the assert statement will not be executed.

Reply:
just took another look at pytest docs

there's a way of making this cleaner which I will add in a moment